### ISSUE TITLE
Support for backward embedding op

### ISSUE THREAD

**Issue by rpavlovicTT** [tenstorrent/tt-forge-fe#670]
Currently, embedding op does not have implementation for backward pass. To update embedding op's weights we need to adjust incoming gradients according to embedding weights dimensions. 

To implement it there are 2 options: 
1. It can be done with a matmul between propagated gradients and sparse matrix generated w.r.t. input sequence. Result of the matmul is resulting gradient to be applied to embedding weights.
2. TTNN has embedding backward op supported. 

Both options require preserving input sequence.

Example of TTNN embedding backward
```
        Example:
            >>> device_id = 0
            >>> device = ttnn.open_device(device_id=device_id)
            >>> batch_size, seq_len, embedding_dim, num_embeddings = 2, 1024, 4096, 3200

            >>> input_shape = (batch_size, seq_len)
            >>> input_index = torch.randint(0, num_embeddings, input_shape)
            >>> input_tensor = ttnn.from_torch(input_index, dtype=ttnn.uint32, device=device)

            >>> weights_shape = (num_embeddings, embedding_dim)
            >>> weights = torch.randn(weights_shape, requires_grad=True)
            >>> weights_ttnn = ttnn.from_torch(weights, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)

            >>> grad_shape = (1, 1, batch_size * seq_len, embedding_dim)
            >>> grad_data = torch.randn(grad_shape, requires_grad=True)
            >>> grad_tensor = ttnn.from_torch(grad_data, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT, device=device)

            >>> output = ttnn.embedding_bw(input_tensor, weights_ttnn, grad_tensor, dtype=ttnn.bfloat16))
```
